### DISCUSSION TITLE
Renderer `\markdownRendererInputVerbatim` no line breaks

### DISCUSSION THREAD

**Discussion by csrmy** [Witiko/markdown#497] [Q&A] [answered]
Hi,
I am new to Tex and LaTex.

While overriding `\markdownRendererInputVerbatim` I figured that if the content is going over the line width, no line break is added automatically when using verbatim or `\inputMinted` (which I am pretty uses verbatim on it's own).

Do you have any idea how to automatically add a line break with verbatim render going over line width ?

Tex file:
``` tex
% Override markdown render of code blocks
\ExplSyntaxOn% enable expl3 syntax
\global\def\markdownRendererInputFencedCode#1#2#3{%
	\catcode`\#=6\relax
		\tl_if_empty:nTF
		{ #2 }
		{ \markdownRendererInputVerbatim{#1} }
		{ \inputminted{#2}{#1} }
	\catcode`\#=12\relax
}
\ExplSyntaxOff% disable expl3 syntax
```

Markdown file:
~~~
```
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
```
~~~

Render:
![image](https://github.com/user-attachments/assets/a60865d1-ac96-47ef-a31a-4b381bcab420)

**Comment by Witiko** [comment, marked as answer]
The minted package has the `breaklines` option that causes overlong lines to break:

![image](https://github.com/user-attachments/assets/b1f6abc5-1a08-464b-a8f5-16aa0d9497b2)

You can use `\setminted{breaklines=true}` in the preamble to enable this. This will only affect code blocks typeset by the minted package, i.e. fenced code blocks. To typeset all code blocks with the minted package, may replace `\markdownRendererInputVerbatim{#1}` with `\inputminted{text}{#1}`.

See also [the documentation][1] of [the minted package][2].

> ```\catcode`\#=6\relax```

In expl3, you can write `\char_set_catcode_parameter:N \#`.

> ```\catcode`\#=12\relax```

In expl3, you can write `\char_set_catcode_other:N \#`.

 [1]: http://mirrors.ctan.org/macros/latex/contrib/minted/minted.pdf
 [2]: https://ctan.org/pkg/minted

**Reply by Witiko** [reply]
For more information about minted, you may wish to ask at https://github.com/gpoore/minted. The markdown package is just a framework in which you can use other LaTeX packages to typeset your markdown documents. While I know a thing or two about minted, it's best to ask the package authors if you have questions that are not specific to the Markdown package. However, if you are not sure if they are related, it's OK to ask here; I appreciate hearing from users and don't mind answering.

**Reply by csrmy** [reply]
Thank you so much @Witiko. Honestly I knew it was not related to the markdown package directly but I was focused on the verbatim rendering from markdown. Indeed using minted is a good solution, and it worked perfectly for me.

Thank you again for your time !
